File: src/main/java/org/slimefunguguproject/bump/core/recipes/BumpRecipeTypes.java
package org.slimefunguguproject.bump.core.recipes;

import org.slimefunguguproject.bump.implementation.Bump;
import org.slimefunguguproject.bump.implementation.BumpItems;

import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType;

/**
 * This class holds all {@link RecipeType} of Bump.
 *
 * @author ybw0014
 */
public final class BumpRecipeTypes {
    private BumpRecipeTypes() {
        throw new IllegalStateException("Utility class");
    }

    public static RecipeType GETGOLD_SPADE = new RecipeType(
        Bump.createKey("getgold_spade"),
        BumpItems.GETGOLD_SPADE
    );
}

File: src/main/kotlin/io/github/slimefunguguproject/bump/implementation/recipes/BumpRecipeTypes.kt
package io.github.slimefunguguproject.bump.implementation.recipes

import io.github.slimefunguguproject.bump.Bump
import io.github.slimefunguguproject.bump.utils.constant.Keys.createKey
import io.github.thebusybiscuit.slimefun4.api.recipes.RecipeType
import io.github.thebusybiscuit.slimefun4.libraries.dough.items.CustomItemStack
import org.bukkit.Material


object BumpRecipeTypes {
    val GET_GOLD_SPADE: RecipeType = RecipeType(
        "get_gold_spade".createKey(),
        CustomItemStack(
            Material.GOLDEN_SHOVEL,
            Bump.localization.getItemName("_RECIPE_GET_GOLD_SPADE"),
            Bump.localization.getItemLore("_RECIPE_GET_GOLD_SPADE")
        )
    )
}
